$OpenBSD: patch-common_util_c,v 1.3 2013/10/02 11:49:00 dcoppa Exp $

commit 381d840a8245c06c88656edc26ed715ad3a53c50
Author: Uli Schlachter <psychon@znc.in>
Date:   Fri Sep 27 16:22:55 2013 +0200

Use $PATH when starting $SHELL

If $SHELL is set to "bash", previously awesome failed to restart itself, because
it could not find "bash". This commit makes awesome use execlp() instead of
execl() which means that $PATH is searched if the started command does not
contain a slash and this problem is fixed.

$SHELL is specified in POSIX and it doesn't seem to require an absolute path
name.

Signed-off-by: Uli Schlachter <psychon@znc.in>

--- common/util.c.orig	Wed Oct  2 12:44:34 2013
+++ common/util.c	Wed Oct  2 12:45:27 2013
@@ -121,8 +121,8 @@ a_exec(const char *cmd)
     if(!shell && !(shell = getenv("SHELL")))
         shell = "/bin/sh";
 
-    execl(shell, shell, "-c", cmd, NULL);
-    fatal("execv() failed: %s", strerror(errno));
+    execlp(shell, shell, "-c", cmd, NULL);
+    fatal("execlp() failed: %s", strerror(errno));
 }
 
 // vim: filetype=c:expandtab:shiftwidth=4:tabstop=8:softtabstop=4:textwidth=80
